scaler/object_storage/object_storage_server.cpp

```
@@ -16,8 +16,12 @@ ObjectStorageServer::~ObjectStorageServer()
     closeServerReadyFds();
 }
 
-void ObjectStorageServer::run(std::string name, std::string port)
+void ObjectStorageServer::run(
+    std::string name, std::string port, std::string log_level, std::string log_format, std::string log_path)
 {
+    assert(!log_level.empty() && !log_format.empty());
```

This should be in the `Logger` class's constructor I think. And maybe raise a `std::invalid_argument` instead of an using assets.

---

i dont think we need this because we have default values for these params. Ill delete this entirely